sympy/physics/mechanics/functions.py

```diff
@@ -729,3 +729,33 @@ def flist_iter():
         unzip = lambda l: list(zip(*l)) if l[0] else [(), ()]
         vel_list, f_list = unzip(list(flist_iter()))
     return vel_list, f_list
+
+
+def _validate_coordinates(coordinates=None, speeds=None, check_duplicates=True,
+                          is_dynamicsymbols=True):
+    # Convert input to iterables
+    if coordinates is None:
+        coordinates = []
+    elif not iterable(coordinates):
+        coordinates = [coordinates]
+    if speeds is None:
+        speeds = []
+    elif not iterable(speeds):
+        speeds = [speeds]
+
+    if check_duplicates:  # Check for duplicates
+        if len(coordinates) != len(set(coordinates)):
+            raise ValueError('Duplicate generalized coordinates found, all '
+                             'generalized coordinates should be unique.')
```

Comment:
If the purpose of this PR is to improve error messaging, do we want to be even more descriptive for the user here? I.e. by specifying which coordinate associated with which joint is duplicated? Or are we happy that this is sufficient? Same goes for the generalised speeds error message below.

Reply:
Yeah, in a way I wanted to do so, but if this is preferred we have to create a loop. And the user can quite easily check the coordinates and speeds parsed to the function.